lib/semantics/expression.cc

```diff
@@ -1437,9 +1438,21 @@ MaybeExpr ExpressionAnalyzer::Analyze(
   return AsMaybeExpr(Expr<SomeDerived>{std::move(result)});
 }
 
-std::optional<ProcedureDesignator>
-ExpressionAnalyzer::AnalyzeProcedureComponentRef(
-    const parser::ProcComponentRef &pcr) {
+static const semantics::WithPassArg *GetPassInfo(
+    const semantics::Symbol &symbol) {
+  if (const auto *binding{symbol.detailsIf<semantics::ProcBindingDetails>()}) {
+    return static_cast<const semantics::WithPassArg *>(binding);
+  } else if (const auto *proc{
+                 symbol.detailsIf<semantics::ProcEntityDetails>()}) {
+    return static_cast<const semantics::WithPassArg *>(proc);
```

Comment:
This cast and the one above aren't necessary, are they?

Reply:
I wanted to be explicit about casting to a base class.  Will remove.